### ISSUE TITLE
Issue with ip6_prefix_list

### ISSUE THREAD

**Issue by vse-github** [fortinet-ansible-dev/ansible-galaxy-fortios-collection#118]
- Galaxy collection "fortinet.fortios" version: 2.1.0 (tested also on 1.1.7)
- FortiOS platform version : 6.4.2

hello, i have an issue to add a ip6_prefix_list through ansible galaxy module **fortinet.fortios.fortios_system_interface**.

It returns 500 error on API even ipv6 address is well configured, but not ipv6 prefix list. Same parameters working through CLI
```
ip6_prefix_list:
  - prefix: "cc::/64"
    autonomous_flag: "disable"
```

**Task example:**
```
  fortinet.fortios.fortios_system_interface:
    vdom: "{{ vdom }}"
    state: "present"
    access_token: "{{ fortios_access_token | default(omit) }}"
    system_interface:
      name: "{{ item.name }}"
      alias: "{{ item.alias }}"
      vdom: "{{ item.vdom }}"
      interface: "{{ item.interface }}"
      vlanid: "{{ item.vlan }}"
      ipv6:
        ip6_address: "cc::1/64"
        ip6_allowaccess: "ping"
        ip6_send_adv: "enable"
        ip6_manage_flag: "enable"
        ip6_prefix_list:
          - prefix: "cc::/64"
            autonomous_flag: "disable"
```

- task log: 

> failed: [fw-2-1] (item={'name': 'C00200100000002', 'vdom': '****', 'vlan': '2', 'ipv6': {'ip6_address': 'cc::1/64', 'ip6_allowaccess': 'ping', 'ip6_send_adv': 'enable', 'ip6_manage_flag': 'enable', 'ip6_prefix_list': [{'prefix': 'cc::/64', 'autonomous_flag': 'disable'}]}}) => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "ansible_loop_var": "item", "changed": false, "item": {"ipv6": {"ip6_address": "cc::/64", "ip6_allowaccess": "ping", "ip6_manage_flag": "enable", "ip6_prefix_list": [{"autonomous_flag": "disable", "prefix": "cc::/64"}], "ip6_send_adv": "enable"}, "name": "C00200100000002", "vdom": "VPN_002001", "vlan": "2"}, "meta": {"build": 1723, "error": -5, "http_method": "POST", "http_status": 500, "name": "interface", "path": "system", "revision": "***", "revision_changed": false, "serial": "FGVM******", "status": "error", "vdom": "***", "version": "v6.4.2"}, "msg": "Error in repo"}

- fortigate api log:

> [httpsd 16624 - 1625059002     info] fweb_debug_init[346] -- New PUT request for "/api/v2/cmdb/system/interface/C00200100000002" from "*****:39139"
> [httpsd 16624 - 1625059002     info] fweb_debug_init[347] -- User-Agent: "Python-urllib/3.7"
> [httpsd 16624 - 1625059002     info] fweb_debug_init[349] -- Handler "api_cmdb_v2-handler" assigned to request
> [httpsd 16624 - 1625059002     info] aps_init_process_vdom[1372] -- initialized process vdom to 'root' (cookie='(null)')
> [httpsd 16624 - 1625059002     info] api_store_parameter[240] -- add API parameter 'vdom' (type=string)
> [httpsd 16624 - 1625059002     info] api_store_parameter[240] -- add API parameter 'name' (type=string)
> [httpsd 16624 - 1625059002     info] api_store_parameter[240] -- add API parameter 'ipv6' (type=object)
> [httpsd 16624 - 1625059002     info] handle_cli_req_v2_vdom[2316] -- attempting to change from vdom "root" to vdom "VPN_002001"
> [httpsd 16624 - 1625059002     info] handle_cli_req_v2_vdom[2319] -- new CMDB API request (vdom='VPN_002001',user='admin')
> [httpsd 16624 - 1625059002     info] api_cmdb_request_init_by_path[1544] -- new CMDB query (path='system',name='interface')
> [httpsd 16624 - 1625059002     info] api_cmdb_request_init_by_path[1573] -- querying CMDB entry (mkey='C00200100000002')
> [httpsd 16624 - 1625059002     info] _api_cmdb_v2_config[1216] -- editing CLI object (append=0, auto_key=0, path=system, name=interface, mkey=C00200100000002, flags=0)
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'name'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-mode'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-mode'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-cert'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-security-level'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-timestamp-delta'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-timestamp-fuzz'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'nd-cga-modifier'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-dns-server-override'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'ip6-address'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'ip6-allowaccess'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'ip6-send-adv'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'ip6-manage-flag'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-other-flag'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-max-interval'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-min-interval'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-link-mtu'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-reachable-time'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-retrans-time'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-default-life'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-hop-limit'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'autoconf'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-upstream-interface'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'ip6-subnet'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'prefix'
> [httpsd 16624 - 1625059002     info] api_set_cmdb_attr[1788] -- 'autonomous-flag'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-relay-service'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-relay-type'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-relay-ip'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-client-options'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-prefix-delegation'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-information-request'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-prefix-hint'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-prefix-hint-plt'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'dhcp6-prefix-hint-vlt'
> [httpsd 16624 - 1625059002     info] api_unset_cmdb_node[1744] -- 'vrrp-virtual-mac6'
> [httpsd 16624 - 1625059002     info] cmdb_save_with_children[279] -- appended main node (nret=0, is_new=0)
> [httpsd 16624 - 1625059002     info] cmdb_save_with_children[279] -- appended main node (nret=0, is_new=0)
> [httpsd 16624 - 1625059002     info] cmdb_save_with_children[279] -- appended main node (nret=-56, is_new=1)
> [httpsd 16624 - 1625059002    error] cmdb_save_with_children[285] -- saving failed for main node: 'ip6-prefix-list' (err=-56)
> 
> [httpsd 16624 - 1625059002    error] cmdb_commit_from_json[1601] -- error saving request object to CLI (-56)
> [httpsd 16624 - 1625059002    error] _api_cmdb_v2_config[1251] -- error editing object (nret=-56)
> [httpsd 16624 - 1625059002  warning] api_return_http_result[699] -- API error -56 raised
> [httpsd 16624 - 1625059002     info] handle_cli_req_v2_vdom[2327] -- returning to original vdom "root"
> [httpsd 16624 - 1625059002     info] fweb_debug_final[248] -- Completed PUT request for "/api/v2/cmdb/system/interface/C00200100000002" (HTTP 500)
> 

Could you help me to solve this issue?

thanks!

**Comment by JieX19** [comment]
Hi @vse-github 

I've reproduced the error and will update you once it's fixed.

Thanks

**Comment by vse-github** [comment]
hello, do you have an ETA for next release including this fix?
thanks

**Comment by JieX19** [comment]
Hi @vse-github 

I am sorry for the late response. It goes to the junk email 😭 . The error was fixed in our latest version. Could you please download the latest fortios ansible galaxy collection using command **ansible-galaxy collection install fortinet.fortios:2.1.2**, and upgrade your device to Fortios 7.0.0. 

Please let me know if it works fine on your side.
Thanks!

**Comment by vse-github** [comment]
hello @JieX19 is this bug could be fixed under 6.4.X branch or Fortios 7.0.0 is mandatory?
thanks

**Comment by JieX19** [comment]
Hi @vse-github Thanks for the feedback! I will open a ticket internally to inform the api team to fix it under 6.4.X branch.

**Comment by JieX19** [comment]
Hi @vse-github I will get back to you once it's fixed.  Thanks!

**Comment by JieX19** [comment]
Hi @vse-github, just got the response from api team, it cannot be fixed in 6.4.x branch, they suggest using the latest version to avoid any other issues due to there're many bugfixes in version 7 and it's more stable.

Thanks for your understanding!
